tests/macros/utils.js

```diff
@@ -191,12 +385,54 @@ function lintHTML(html, fragment = true) {
     }
 }
 
+/**
+ * Reads a generic fixture file from the `fixtures` directory.
+ *
+ * @param {string|string[]} filePath
+ *        A path to a file relative to the `fixtures` directory.
+ * @param {string|null} [encoding]
+ *        The file encoding. Set to `null` to load the file as a binary `Buffer`.
+ *
+ * @returns {string}
+ */
+function readFixture(filePath, encoding = 'utf-8') {
+    if (!Array.isArray(filePath)) {
+        filePath = [filePath];
+    }
+    let absolutePath = path.resolve(__dirname, 'fixtures', ...filePath);
+    return fs.readFileSync(absolutePath, typeof encoding === 'string' ? encoding : undefined);
+}
+
+/**
+ * Reads a JSON fixture file from the `fixtures` directory.
+ *
+ * @template T The schema of the JSON payload.
+ * @param {...string} filePath
+ *        A path to a file relative to the `fixtures` directory.
+ *
+ *        The `.json` file extension can be omitted.
+ *
+ * @returns {T} The parsed JSON payload.
```

Comment:
This&nbsp;is&nbsp;added so&nbsp;that the&nbsp;type cast gets performed automatically.